spring-boot-modules/spring-boot-crud/src/test/java/com/baeldung/entity_dto_differences/entity/UserUnitTest.java

```diff
@@ -0,0 +1,65 @@
+package com.baeldung.entity_dto_differences.entity;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UserUnitTest {
+
+    @Test
+    public void whenUserInitialized_thenCorrect() {
+
+        Book book1 = new Book("Book1", "Author1");
+        Book book2 = new Book("Book2", "Author2");
+
+        User user = new User("John", "Doe", "123 Main St", Arrays.asList(book1, book2));
+
+        assertNotNull(user);
+        assertEquals("John", user.getFirstName());
+        assertEquals("Doe", user.getLastName());
+        assertEquals("123 Main St", user.getAddress());
+        assertEquals(2, user.getBooks().size());
+    }
+
+    @Test
+    public void whenGettersAndSettersAreCalled_thenCorrect() {
+
+        User user = new User();
+        user.setId(1L);
+        user.setFirstName("Jane");
+        user.setLastName("Doe");
+        user.setAddress("456 Oak St");
+        List<Book> books = Arrays.asList(new Book("Book3", "Author3"), new Book("Book4", "Author4"));
+        user.setBooks(books);
+
+        assertEquals(1L, user.getId());
+        assertEquals("Jane", user.getFirstName());
+        assertEquals("Doe", user.getLastName());
+        assertEquals("456 Oak St", user.getAddress());
+        assertEquals(2, user.getBooks().size());
+    }
+
+    @Test
+    public void whenGetNameOfMostOwnedBook_thenCorrect() {
+
+        Book book1 = new Book("Book1", "Author1");
+        Book book2 = new Book("Book2", "Author2");
+        Book book3 = new Book("Book2", "Author3");
+
+        User user = new User("John", "Doe", "123 Main St", Arrays.asList(book1, book2, book3));
+
+        assertEquals("Book2", user.getNameOfMostOwnedBook());
+    }
+
+    @Test
+    public void whenGetNameOfMostOwnedBook_withNoBooks_thenCorrect() {
```

Comment:
The format we require for test names is given_when_then if there are givens, or when_then if there are no givens

This for example should be named something like `givenUserWithNoBooks_whenGetNameOfMostOwnedBook_thenReturnNull`

I am also not a fan of this `thenCorrect` naming, it does not make it clear what "correct" means. Let's be specific in what our "then" expectations are please. Same in all your other tests.